server/src/main/java/org/cloudfoundry/identity/uaa/zone/UserConfigValidator.java

```diff
@@ -9,10 +9,19 @@ private UserConfigValidator() {
     }
 
     public static void validate(UserConfig config) throws InvalidIdentityZoneConfigurationException {
-        Set<String> allowedGroups = (config == null) ? null : config.resultingAllowedGroups();
-        if ((allowedGroups != null) && (allowedGroups.isEmpty())) {
+        if (config == null) {
+            return;
+        }
+
+        Set<String> allowedGroups = config.resultingAllowedGroups();
+        if (allowedGroups != null && allowedGroups.isEmpty()) {
             String message = "At least one group must be allowed";
             throw new InvalidIdentityZoneConfigurationException(message);
         }
+
+        long maxUsers = config.getMaxUsers();
+        if (maxUsers < -1 || maxUsers == 0) {
+            throw new InvalidIdentityZoneConfigurationException("Maximum number of users in the zone is invalid, either use -1 or a value more than 0.");
+        }
```

Comment:
Another option is to make the default `maxUsers` value `0` and `0` value to mean unlimited. Then this condition for example can be changed to `maxUsers < 0`. I think doing so will result in more readable code.

Reply:
This is the validator for the rest interface. I dont see this as a complicated logic. It should simply help to have clean values. 
We use -1 for many other numbers which the meaning of "do nothing", e.g. token or password policy. Here -1 is always no check so I would like to have -1 because we have it already.

Allowing no user maxUsers == 0 is something if you dont want it I can remove, but then no user in a zone could be created... 

So would you agree with 

if (maxUsers < -1) {
            throw new InvalidIdentityZoneConfigurationException("Maximum number of users in the zone is invalid, either use -1 or a value.");
        }

Reply:
> We use -1 for many other numbers which the meaning of "do nothing"

OK, then I am find to keep it as it is.

> Allowing no user maxUsers == 0 is something if you dont want it I can remove

No, you should keep it as the default will stay as `-1`.

Reply:
thanks close this